meilisearch-release-changelog/v0.30.0.md

```diff
@@ -0,0 +1,240 @@
+# v0.30.0 release changelogs
+
+<!-- The following line should NOT be put in the official release changelogs -->
+⚠️ Since this is a release candidate (RC), we do NOT recommend using it in a production environment. Something not working as expected? We welcome [bug reports](https://github.com/meilisearch/meilisearch/issues/new/choose) and [feedback about new features](https://github.com/meilisearch/product/discussions).
+
+This release introduces index deployment with zero downtime and adds task management features. It also brings the capability to navigate search results by page selection.
+
+Thanks to Hacktoberfest, tons of updates and bug fixes have been done by the fantastic Meilisearch community. You were more present than ever and words cannot do justice tojust how grateful we are for your support. A big thank you from the whole Meilisearch team! ❤️
+
+<!-- The following line should NOT be put in the official release changelogs -->
+📖 Meilisearch's official documentation will be ready for the official v0.30.0 release. Meanwhile, work-in-progress documentation is available [here]().
+
+# New features and improvements 🔥
+
+## Improve search result navigation: create numbered pagination interfaces
+
+You can now use `page` and `hitsPerPage` to fetch the total number of pages and results for a query. These new search parameters can be used in the place of `offset` and `limit` when creating interfaces that require knowing the exhaustive number of search results.
+
+The following example fetches the second page of results for a given query:
+
+```bash
+curl \
+  -X POST 'http://localhost:7700/indexes/movies/search' \
+  -H 'Content-Type: application/json' \
+  --data-binary '{ "q": "shifu", "page": "2", "hitsPerPage": "10" }'
+```
+
+```json
+{
+    "hits": [
+        // … 10 hits
+    ],
+    // …
+    "page": 2,
+    "hitsPerPage": 10,
+    "totalHits": 2100,
+    "totalPages": 210
+}
+```
+
+Done in #2601 by @ManyTheFish.
```

Comment:
Opening new discussion regarding [this comment](https://github.com/meilisearch/product/discussions/335#discussioncomment-4041129)
This is currently not easy to understand for people currently doing pagination with Meilisearch before this feature was introduced.